plugins/woocommerce/src/Blocks/TemplateOptions.php

```diff
@@ -0,0 +1,37 @@
+<?php
+namespace Automattic\WooCommerce\Blocks;
+
+use Automattic\WooCommerce\Blocks\Options;
+
+/**
+ * TemplateOptions class.
+ *
+ * @internal
+ */
+class TemplateOptions {
+
+	/**
+	 * Initialization method.
+	 */
+	public function init() {
+		add_action( 'after_switch_theme', array( $this, 'check_should_use_blockified_product_grid_templates' ), 10, 2 );
+	}
+
+	/**
+	 * Checks the old and current themes and determines if the "wc_blocks_use_blockified_product_grid_block_as_template"
+	 * option need to be updated accordingly.
+	 *
+	 * @return void
+	 */
+	public function check_should_use_blockified_product_grid_templates() {
+		$option_name           = Options::WC_BLOCK_USE_BLOCKIFIED_PRODUCT_GRID_BLOCK_AS_TEMPLATE;
+		$should_use_blockified = wc_bool_to_string( wc_current_theme_is_fse_theme() );
+
+		// We don't need to do anything if the user switched to a classic theme or if the option is already set to true.
+		if ( ! $should_use_blockified || get_option( $option_name ) === 'true' ) {
```

Comment:
When we are updating the option we are using `wc_bool_to_string` which turns `true` values to `"yes"` and `false` values to `"no"` but when we are operating on the value of this option (line 31) we are comparing against `true`? Is this correct?

Just curious if we have incorrectly assumed `wc_bool_to_string` would turn `true` into `"true"` based on this code here on line 31.

```suggestion
		if ( ! $should_use_blockified || get_option( $option_name ) === wc_bool_to_string( true ) ) {
```

Reply:
An alternative to the above suggestion would be to use our [`wc_string_to_bool` function](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/wc-formatting-functions.php#L25)

```suggestion
		if ( ! $should_use_blockified || wc_string_to_bool( get_option( $option_name ) ) ) {
```

Reply:
Your suggestion makes sense, merging it. :slightly_smiling_face: